src/query/Index.cpp

```diff
@@ -230,7 +230,7 @@ std::vector<VertexID> Index::getVerticesInsideBox(const mesh::Vertex &centerVert
 
   const auto &          rtree = _pimpl->getVertexRTree(*_mesh);
   std::vector<VertexID> matches;
-  rtree->query(bgi::intersects(searchBox) and bg::index::satisfies([&](size_t const i) { return bg::distance(centerVertex, _mesh->vertices()[i]) <= radius; }),
+  rtree->query(bgi::intersects(searchBox) and bg::index::satisfies([&](size_t const i) { return bg::distance(centerVertex, _mesh->vertices()[i]) < radius; }),
```

Comment:
Numerically, there is anyway no real difference between "<=" and "<", no? Then, I don't understand the motivation for this  change.
Shouldn't there be a

https://github.com/precice/precice/blob/fbe44d289a9d257d0e8ab1e226cf75f35eef5e23/src/math/differences.hpp#L89-L93

anyway?

Reply:
Hitting the 'exact' zero was no coincidence in the first place. Hence, there is a difference in both versions. To some extend, the problem was solved already with #1736 . I will subtract the numerical difference from the computed radius as well to make it a bit more robust, but this shouldn't really be necessary IMO.